src/Tools/SchemaValidator.php

```diff
@@ -276,6 +277,22 @@ public function validateClass(ClassMetadata $class): array
                     }
                 }
             }
+
+            if ($assoc->isIndexed()) {
+                $joinColumns = array_column($targetMetadata->getAssociationMappings(), 'joinColumns');
+                $joinColumns = array_column($joinColumns, 0);
+                $joinColumns = array_column($joinColumns, 'name');
+
```

Comment:
I suggest using a `foreach` loop instead of several successive `array_columns`, allowing to loop once

Reply:
I'm not sure I follow. `$targetMetadata->getAssociationMappings()` returns a multi-dimensional array. I'll have to have [probably three] nested foreach loops to gather all the data anyway.